### PR TITLE
OTP 26 compat: fix ssl options on Tesla request

### PR DESCRIPTION
**Pull request by tfiedlerdejanze**
I have tested this with localstack against sched on OTP 26 and confirmed it's working.

### CODE REVIEW

**Comment by tfiedlerdejanze** [comment]
@aaronrenner @edennis looks like the index we're using to run the tests is full :(

File: lib/algolia.ex [hunk 1]
```diff
@@ -168,11 +167,19 @@ defmodule Algolia do
...
           recv_timeout: 30_000 * (curr_retry + 1),
-          ssl_options: [{:versions, [:"tlsv1.2"]}]
+          ssl_options: [
+            {:versions, [:"tlsv1.2"]},
+            {:cacerts, :public_key.cacerts_get()},
+            {:customize_hostname_check,
+             [
+               match_fun: :public_key.pkix_verify_hostname_match_fun(:https)
+             ]}
+          ]
```
Comment (aaronrenner) [inline lib/algolia.ex]: Do we need to bump the minimum version of any of our dependencies to support these new options?

Also, could this be a keyword list?

```elixir
[
  versions: [:"tlsv1.2"],
  cacerts: :public_key.cacerts_get()},
  customize_hostname_check:
             [
               match_fun: :public_key.pkix_verify_hostname_match_fun(:https)
             ]
]
```

Comment (tfiedlerdejanze) [reply lib/algolia.ex]: I dont think any of these are new: https://www.erlang.org/doc/apps/ssl/notes

Elixir 1.15.7 uses `ssl: 10.9`

**Comment by aaronrenner** [comment]
It looks like there are a couple issues...

1. The test suite appears to not be cleaning up after itself and we're running out of indexes. We should probably address this separately.
2. I'm seeing a couple SSL related warnings in our test output:
```
Warning: 13:08:56.520 [warning] Description: ~c"Server authenticity is not verified since certificate path validation is not enabled"
     Reason: ~c"The option {verify, verify_peer} and one of the options 'cacertfile' or 'cacerts' are required to enable this."
```

This will be a great change, but we should probably get these resolved first.

File: lib/algolia.ex (hunk 1 above)
Comment (tfiedlerdejanze) [reply lib/algolia.ex]: possible that it supports keyword lists, i'd have to try. Do you mind the list of tuples?

Comment (aaronrenner) [reply lib/algolia.ex]: The list of tuples is fine. I'm just thinking the formatter would most likely change it to a keyword list syntax.

**Review by aaronrenner — APPROVED** [review]

**Pull request merged**